brontology/relation_extraction/delegator.py

```diff
@@ -0,0 +1,66 @@
+from enum import Enum, Flag, auto, IntEnum
+# noinspection PyUnresolvedReferences
+from typing import Union, Optional, Callable
+
+from spacy.parts_of_speech import VERB
+from spacy.tokens import Span, Token
+
+from brontology.relation_extraction.relation_types.passive_voice import extract_passive_voice_relation
+from brontology.relation_extraction.types import TokenRelation
+from brontology.utils.color import A
+
+PROCESSORS: list[Callable[[Token], TokenRelation | None]] = list()
+
+
+def add_processor(processor: Callable[[Token], TokenRelation | None]):
+    """Adds `processor` to `PROCESSORS`.
+    The processor should take a main verb and return a `TokenRelation`
+     if the phrase of the verb matches the type of the processor, or `None` if not."""
+    PROCESSORS.append(processor)
+
+
+def get_main_verbs(span: Span) -> list[Token]:
+    """Returns all the main verbs of the `span`."""
+    tokens = list()
+    for token in span:
+        if token.pos == VERB:
+            tokens.append(token)
+    return tokens
+
+
+def extract_relation(verb: Token) -> TokenRelation:
+    """Extracts the relation formed by the `verb` using the defined `PROCESSORS`."""
+    for processor in PROCESSORS:
+        relation: Optional[TokenRelation] = processor(verb)
+        if relation is not None:
+            return relation
+    raise ValueError("Could not extract relation.")
+
+
+add_processor(extract_passive_voice_relation)
+
+if __name__ == '__main__':
```

Comment:
Code, der hinter so einer `if __name__ == '__main__':`-Klasel steckt, muss in der Regel nicht gereviewed werden, ist allerdings lesenswert, da sich dort oft eine Demonstration des Moduls befindet.